Lib/test/test_ntpath.py

```diff
@@ -1095,6 +1095,22 @@ def test_isfile_driveletter(self):
             raise unittest.SkipTest('SystemDrive is not defined or malformed')
         self.assertFalse(os.path.isfile('\\\\.\\' + drive))
 
+    @unittest.skipUnless(hasattr(os, 'pipe'), "need os.pipe()")
+    def test_isfile_anonymous_pipe(self):
+        pr, pw = os.pipe()
+        try:
+            self.assertFalse(ntpath.isfile(pr))
+        finally:
+            os.close(pr)
+            os.close(pw)
+
+    @unittest.skipIf(sys.platform != 'win32', "windows only")
+    def test_isfile_named_pipe(self):
+        import _winapi
+        named_pipe = f'//./PIPE/python_isfile_test_{os.getpid()}'
+        _winapi.CreateNamedPipe(named_pipe, _winapi.PIPE_ACCESS_INBOUND, 0, 1, 0, 0, 0, 0)
+        self.assertFalse(ntpath.isfile(named_pipe))
```

Comment:
`CreateNamedPipe()` returns a handle to the server end of the pipe. Pipe files in the named-pipe filesystem are reference counted by handles to the server end and the client end. When handle `h` is closed, `named_pipe` is automatically unlinked from the filesystem root directory "\\\\.\\PIPE\\" (i.e. "\\Device\\NamedPipe\\").

```suggestion
        h = _winapi.CreateNamedPipe(named_pipe,
                                    _winapi.PIPE_ACCESS_INBOUND,
                                    0, 1, 0, 0, 0, 0)
        try:
            self.assertFalse(ntpath.isfile(named_pipe))
        finally:
            _winapi.CloseHandle(h)
```